whisper/utils.py

```diff
@@ -85,3 +85,27 @@ def write_srt(transcript: Iterator[dict], file: TextIO):
             file=file,
             flush=True,
         )
+
+def write_csv(transcript: Iterator[dict], file: TextIO):
+    """
+    Write a transcript to a file in CSV format containing lines like:
+    <startTime-in-integer-milliseconds>, <endTime-in-integer-milliseconds>, <transcript-including-commas>
+    
+    Using integer milliseconds as start and end times means there's no chance of interference from an environment
+    setting a language encoding that causes the decimal in a floatinng point number to appear as a comma; also is
+    faster and more efficient to parse & store, e.g., in a C++ application.
+
+    Example usage 
+        from pathlib import Path
+        from whisper.utils import write_csv
+
+        result = transcribe(model, audio_path, temperature=temperature, **args)
+
+        # save CSV -- see transcribe.py
+        with open(os.path.join(output_dir, os.path.basename(audio_path) + ".csv"), "w", encoding="utf-8") as csv:
+            write_csv(result["segments"], file=csv)
+    """    
+    for segment in transcript:
+        print(round(1000 * segment['start']), file=file, end = ', ')
+        print(round(1000 * segment['end']),   file=file, end = ', ')  
```

Comment:
With the name of the ticket, I was excited to get millisecond resolution of the transcription, to help with speaker recognition weaving.  I notice that for the MS resolution you just multiply by 1000.

Any way of getting segment['end'] to be in ms, and have a switch to round if that becomes inconvenient?

Reply:
The purpose of this patch was to ease parsing, efficiency, storage, etc., for an application that already uses and requires millisecond timestamps for events. This includes potential rounding differences going from milliseconds, to H:M:S.MM especially when these need to get reconverted back to milliseconds.  There's also the internationalization issues with some localities using "." for decimal and others "," -- milliseconds sidesteps all that, alongside increased parsing speed, cross-platform rounding accuracy, etc.

Yes, of course I would prefer getting more precise timestamps back from the internals of whisper. Yet the multiplication by 1000 isn't such a bad heuristic given that the returned data appears to have accuracy within 10 milliseconds. If you look at whisper data, at least for my chosen transcription parameters using the "medium" model,. you see, e.g., 

```
17
00:02:05,240 --> 00:02:12,120
keeping the sample relationship between PNs and C++. And this topic, especially, is my

18
00:02:12,120 --> 00:02:24,920
I made a program in superfast, more superfast, using better nets, and we will see what this

19
00:02:24,920 --> 00:02:31,440
program makes and some details.

20
00:02:31,440 --> 00:02:41,240
So I assume many people here are related with the concept of design of systems, and every

21
00:02:41,240 --> 00:02:50,640
day more specific and more complicated systems are rising with the new advancing technology

22
00:02:50,640 --> 00:02:52,840
and so on.

23
00:02:52,840 --> 00:02:59,720
So there is increased emphasis on speed in the analysis and design of this system.

24
00:02:59,720 --> 00:03:10,040
And it's important to make focus in the design of the system because understanding system

```

So at least for me I'm seeing accuracy in 10's of milliseconds for CSV output:

```
125240, 132120, "keeping the sample relationship between PNs and C++. And this topic, especially, is my"
132120, 144920, "I made a program in superfast, more superfast, using better nets, and we will see what this"
144920, 151440, "program makes and some details."
151440, 161240, "So I assume many people here are related with the concept of design of systems, and every"
161240, 170640, "day more specific and more complicated systems are rising with the new advancing technology"
170640, 172840, "and so on."
172840, 179720, "So there is increased emphasis on speed in the analysis and design of this system."
179720, 190040, "And it's important to make focus in the design of the system because understanding system"

```